main.tf

```
@@ -1,15 +1,91 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled                   = module.this.enabled
+  prometheus_policy_enabled = local.enabled && var.prometheus_policy_enabled
+}
+
+resource "aws_grafana_workspace" "this" {
+  count = local.enabled ? 1 : 0
+
+  name        = module.this.id
+  description = "Amazon Managed Grafana for ${module.this.id}"
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  account_access_type      = "CURRENT_ACCOUNT"
+  authentication_providers = var.authentication_providers
+  permission_type          = var.permission_type
+  role_arn                 = aws_iam_role.this[0].arn
+  data_sources             = []
+
+  dynamic "vpc_configuration" {
+    for_each = var.vpc_configuration != null ? [1] : []
+    content {
+      security_group_ids = var.vpc_configuration.security_group_ids
+      subnet_ids         = var.vpc_configuration.subnet_ids
+    }
   }
+
+  tags = module.this.tags
 }
 
+resource "aws_iam_role" "this" {
+  count = local.enabled ? 1 : 0
 
+  name = module.this.id
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "grafana.amazonaws.com"
+        }
+      },
+    ]
+  })
+
+  # Allow this role to assume other roles.
+  # Used for cross-account access
+  dynamic "inline_policy" {
+    for_each = length(var.additional_allowed_roles) > 0 ? [1] : []
+    content {
+      name = "${module.this.id}-allowed-roles"
```

Not using null label properly. 
```suggestion
      name = "${module.this.id}-allowed-roles"
```

---

replaced with separate null-label modules for creating sub-labels